### ISSUE TITLE
Feature Center

### ISSUE THREAD

**Issue by mbogaerts** [eXpandFramework/eXpand#774]
𝗤𝘂𝗲𝘀𝘁𝗶𝗼𝗻
Is the feature center stil somewhere to be found?
i'm struggling to get expand to work in a new xaf 20.1 project.

e.g. the excel importers behavior is not as it should be and it is a struggle to figure out missing dependencies etc.

𝗛𝗼𝘄 𝘄𝗲 𝗽𝗿𝗶𝗼𝗿𝗶𝘁𝗶𝘇𝗲 𝗶𝘀𝘀𝘂𝗲𝘀
We will try to answer all questions that do not require research within 24hr.
To prioritize cases that require research we use the following labels in order.

01. ❇️ Exclusive-Services
02. ❤ Bronze Sponsor
03. ❤ Sponsor
04. ❤ Backer
05. Installation
06. ShowStopper
07. Nuget
08. Contribution
09. BreakingChange
10. ReproSample
11. Deployment
12. Must-Have

For all other issues the posting time is respected.

**Comment by apobekiaris** [comment]
FeatureCenter still exists in git history however it is removed from latest versions as there are no tests for it so it only adds confusion. In any case the ExcelImporter is a module published recently so nothing can be found in the old FeatureCenter. Finally currently the framework is only distributed from Nuget and if you post details on your problem and you use nuget instead of direct references happy to share my quick opinion. 

Research requires a sample and is prioritized as the bot already notified you.

**Comment by mbogaerts** [comment]
Please have a quick look at the screencast... https://www.screencast.com/t/3n91vcDsH4Q

it seems to be missing out on some functions like resolving the properties/excel sheets
I have have seen this in the video's you made.

**Comment by apobekiaris** [comment]
do you have both the Excel.Win module and the System.Win module ?

**Comment by apobekiaris** [comment]
if you use nugets and the ExcelImporterWinModule is registered this should be already the case

```ps1
Get-NugetPackageDependencies eXpandExcelImporter.Win

Id                         Include Exclude VersionRange
--                         ------- ------- ------------
ExcelDataReader.DataSet    {}      {}      [3.6.0, )
eXpandExcelImporter        {}      {}      [20.1.701.4, )
eXpandLib                  {}      {}      [20.1.701.4, )
eXpandSystemWin            {}      {}      [20.1.701.4, )
eXpandValidationWin        {}      {}      [20.1.701.4, )
JetBrains.Annotations      {}      {}      [2020.1.0, )
System.IO.Compression      {}      {}      [4.3.0, )
WindowsBase                {}      {}      [4.6.1055, )
Xpand.Extensions.Reactive  {}      {}      [2.202.48.2, )
Xpand.XAF.Modules.Reactive {}      {}      [2.202.48.2, )
```

**Comment by mbogaerts** [comment]
added this to the winmodule did the trick
this.RequiredModuleTypes.Add(typeof(Xpand.ExpressApp.ExcelImporter.Win.ExcelImporterWinModule));

**Comment by apobekiaris** [comment]
makes sense :) to use a XAF module u have to register at least the platform specific the agnostic is already registered as a dependency to the win.

**Comment by mbogaerts** [comment]
i do however find it difficult to get started...
my approach was install via nuget
1) agnostic
2) win in windows projects 
3) web in web projects
must have missed something there then

**Comment by apobekiaris** [comment]
agnostic is not required but does not hurt. On each nuget installation a readme popup for further instructions like the line u missed,

So can u please explain what was hard to follow?

**Comment by mbogaerts** [comment]
I'll get back to you when i have some time to figure out where i messed up... POC this afternoon so need to get things working :-)
onto the next iussue: Reentrancy or cross thread operation detected. Use ThreadSafeDataLayer or find the cause of the cross thread access to the SimpleDataLayer instance from your code as described ...

**Comment by apobekiaris** [comment]
the excelimporter is doind async operations u need to use a ThreadSafeDatalyer I believe is pretty straight fordard message

e.g. note the last argument
```cs
args.ObjectSpaceProvider=new XPObjectSpaceProvider(new ConnectionStringDataStoreProvider(args.ConnectionString),true);
```

web use this thread safe layer by default but win is not

**Comment by mbogaerts** [comment]
Lost touch with xaf and expand for a while now... just came back into the picture a few days ago... under pressure to get the poc ready :-P
I believe is pretty straight fordard message => yes it is 
thnx for the assistance

**Comment by apobekiaris** [comment]
welcome, again I remind u that I only support nuggets currently as there is logic in them and the installer is now removed.

**Comment by mbogaerts** [comment]
yep i know... was a bit confused there as the docs refer to the installer and at first i could not find a lot of info on how to get started with the nugets... maybe it's just me rushing into thing
i'll get back to you on this for sure later on

**Comment by expand** [comment]
Closing issue for age. Feel free to reopen it at any time.

.Thank you for your contribution.
